spec/models/book_spec.rb

```diff
@@ -5,35 +5,76 @@
-    it { is_expected.to validate_presence_of(:title) }
-    it { is_expected.to validate_presence_of(:genre) }
-    it { is_expected.to validate_presence_of(:language) }
-    it { is_expected.to validate_presence_of(:edition) }
-    it { is_expected.to validate_presence_of(:place) }
-    it { is_expected.to validate_presence_of(:year) }
+  describe 'Rails validations' do
+    context 'when validations are needed' do
+      it { is_expected.to define_enum_for(:language).with_values(portuguese: 0, english: 1, spanish: 2) }
+      it { is_expected.to validate_presence_of(:title) }
+      it { is_expected.to validate_presence_of(:genre) }
+      it { is_expected.to validate_presence_of(:language) }
+      it { is_expected.to validate_presence_of(:edition) }
+      it { is_expected.to validate_presence_of(:place) }
+      it { is_expected.to validate_presence_of(:year) }
+      it { is_expected.to validate_numericality_of(:year) }
+      it { is_expected.to validate_length_of(:title).is_at_least(2).is_at_most(50) }
+      it { is_expected.to validate_length_of(:genre).is_at_least(2).is_at_most(50) }
+      it { is_expected.to validate_length_of(:edition).is_at_least(2).is_at_most(50) }
+      it { is_expected.to validate_length_of(:place).is_at_least(2).is_at_most(50) }
+    end
   end
 
-  context 'with relationships' do
-    it { is_expected.to belong_to(:author) }
-    it { is_expected.to belong_to(:publisher) }
+  describe 'relationships' do
+    context 'when relationships are defined' do
+      it { is_expected.to belong_to(:author) }
+      it { is_expected.to belong_to(:publisher) }
+    end
   end
 
-  context 'when the book has the default attributes' do
-    it { expect(book).to be_valid }
+  describe 'book resources' do
+    context 'when the book has the valid attributes' do
+      it { expect(book).to be_valid }
+    end
+
+    context 'when the book has the invalid attributes' do
+      let(:long_str) { Faker::Name.initials(number: 51) }
+      let(:invalid_year) { [rand(2..1499), rand((Time.zone.now.year + 1)..3000)].sample }
 
-    it 'must be valid' do
-      book.language = :spanish
-      expect(book).to be_valid
+      before do
+        book.title = long_str
+        book.genre = long_str
+        book.edition = long_str
+        book.place = long_str
+        book.year = invalid_year
+        book.save
+      end
+
+      it 'generates the following errors' do
+        expect(book.errors.messages[:title]).to eq ['is too long (maximum is 50 characters)']
+        expect(book.errors.messages[:genre]).to eq ['is too long (maximum is 50 characters)']
+        expect(book.errors.messages[:edition]).to eq ['is too long (maximum is 50 characters)']
+        expect(book.errors.messages[:place]).to eq ['is too long (maximum is 50 characters)']
+        expect(book.errors.messages[:year]).to eq ['must be in 1500..2022']
+        expect(book.valid?).to be false
```

Comment:
Se vc já começar verificando se é válido, já vai conseguir ter as mensagens de validação no `errors`:
```suggestion
        expect(book.valid?).to be false
        expect(book.errors.messages[:title]).to eq ['is too long (maximum is 50 characters)']
        expect(book.errors.messages[:genre]).to eq ['is too long (maximum is 50 characters)']
        expect(book.errors.messages[:edition]).to eq ['is too long (maximum is 50 characters)']
        expect(book.errors.messages[:place]).to eq ['is too long (maximum is 50 characters)']
        expect(book.errors.messages[:year]).to eq ['must be in 1500..2022']
```

Reply:
Pra resolver a issue do build, cria um:
```
let(:errors_message) { 'is too long (maximum is 50 characters)' }
```